core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,828 @@
+                    objects[2].date()))
+        )
+
+        report_invalid_state_interaction_values_part_3 = (
+            invalid_exps_with_errored_state_interaction_values_part_3
+            | 'Show info for state interaction part 3' >> beam.Map(
+                lambda objects: job_run_result.JobRunResult.as_stderr(
+                   f'The id of exp is {objects[0]}, ' +
+                   f'created on {objects[2]}, and the state interaction'
+                   + f' part 3 erroneous data are {objects[1]}'
+                )
+            )
+        )
+
+        invalid_exps_with_errored_states_values = (
+            combine_exp_ids_and_states
+            | 'Get invalid state values' >> beam.Map(
+                lambda objects: (
+                    objects[0],
+                    self.filter_invalid_state_values(objects[1]), objects[2]))
+            | 'Remove empty values for state values' >> beam.Map(
+                lambda objects: (
+                    objects[0], self.remove_empty_values(objects[1]),
+                    objects[2].date()))
+        )
+
+        report_invalid_state_values = (
+            invalid_exps_with_errored_states_values
+            | 'Show info for state values' >> beam.Map(
+                lambda objects: job_run_result.JobRunResult.as_stderr(
+                   f'The id of exp is {objects[0]}, ' +
+                   f'created on {objects[2]}, and the state' +
+                   f' erroneous data are {objects[1]}'
+                )
+            )
+        )
+
+        return (
+            (
+                report_number_of_exps_queried,
+                report_invalid_state_rte_values,
+                report_invalid_state_interaction_values_part_1,
+                report_invalid_state_interaction_values_part_2,
+                report_invalid_state_interaction_values_part_3,
+                report_invalid_state_values
+            )
+            | 'Combine results' >> beam.Flatten()
+        )
+
+    def filter_invalid_state_rte_values(
+        self, states_dict: dict[str, state_domain.State]) -> list[dict]:
+        """Returns the errored state RTE values.
+
+        Args:
+            states_dict: dict[str, State]. The dictionary containing
+                state name as key and State object as value.
+
+        Returns:
+            states_with_values: list[dict]. The list of dictionaries
+            containing the errored values.
+        """
+        states_with_values = []
+        for key, value in states_dict.items():
+            rte_components_errors = []
+            rte_components = (html_cleaner.get_rte_components(
+                value.content.html))
+            if len(rte_components) > 0:
+                for rte_component in rte_components:
+                    if rte_component['id'] == 'oppia-noninteractive-image':
+                        if len(rte_component['customization_args']
+                        ['caption-with-value']) > 160:
```

Comment:
Please before every individual validation add a short comment referencing what we actually validate (feel free to use some kind of shortcuts for this). Then in the issue description list all of these things so that we can easily validate that everything is checked by this job.

Reply:
Done